### PR TITLE
Graph and dav become unusable after executing a combination of queries #460

### PR DESCRIPTION
**Pull request by formalhaut69**
### Description of the Change
A collection of layers view fixes. To resolve incorrect access of attributes.

### CODE REVIEW

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java [hunk 1]
```diff
@@ -1999,10 +2000,16 @@ private void recalculateLayerVisibilities() {
      */
     private int recalculateLayerMask(final GraphElementType elementType, final int elementId) {
         int bitmask = 0;
-        if (elementType == GraphElementType.VERTEX && vertexLayerMaskAttribureId >= 0 && vertexLayerVisibilityAttributeId >= 0) {
-            bitmask = getIntValue(vertexLayerMaskAttribureId, elementId);
-        } else if (elementType == GraphElementType.TRANSACTION && transactionFilterBitmaskAttrId >= 0 && transactionLayerVisibilityAttributeId >= 0) {
-            bitmask = getIntValue(transactionFilterBitmaskAttrId, elementId);
+        if (elementType == GraphElementType.VERTEX) {
+            vertexLayerMaskAttribureId = GraphManager.getDefault().getActiveGraph().getReadableGraph().getAttribute(GraphElementType.VERTEX, LAYER_MASK_ATTRIBUTE_LABEL);
```
Comment (cygnus-x-1) [inline CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java:2004]: Whenever you get a `ReadableGraph` or `WritableGraph` you should ensure that you release the lock when you're done with it. In this case i'd suggest something like:
```
final readableGraph = GraphManager.getDefault().getActiveGraph().getReadableGraph();
try {
    vertexLayerMaskAttributeId = VisualConcept.VertexAttrtibute.LAYER_MASK.get(readableGraph);
} finally {
    readableGraph.release();
}
```

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java [hunk 2]
```diff
@@ -2151,7 +2158,8 @@ private void updateLayerMask(final int attributeId, final GraphElementType eleme
         final int bitmask = recalculateLayerMask(elementType, elementId);
 
         if (elementType == GraphElementType.VERTEX) {
-
+            vertexLayerVisibilityAttributeId = GraphManager.getDefault().getActiveGraph().getReadableGraph()
```
Comment (cygnus-x-1) [inline CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java:2161]: `VisualConcept.VertexAttrtibute.LAYER_MASK.get(readableGraph)`

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java (hunk 1 above)
Comment (cygnus-x-1) [reply CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java:2004]: Also, `StoreGraph` is a `GraphWriteMethods`... so couldn't you just write `vertexLayerMaskAttributeId = VisualConcept.VertexAttrtibute.LAYER_MASK.get(this)`?

Comment (cygnus-x-1) [reply CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java:2004]: Further to this, with context managers available as of Java 9? (8? 7?) could we update the graph API to allow:
```
try (graph.getReadableGraph()) {
    ///do stuff...
}
```

**Comment by arcturus2** [comment]
@formalhaut69 is this PR still relevant or should it be closed?

**Comment by aldebaran30701** [comment]
> @formalhaut69 is this PR still relevant or should it be closed?

see #528  - it may close this PR

**Pull request closed without merging**